src/Support/Traits/HandlesQueryFilter.php

```diff
@@ -17,11 +17,16 @@
      */
     protected function injectFilter(ArgumentValue $argument, array $filter)
     {
+        $argName = $argument->getArgName();
         $key = $this->queryFilterKey($argument);
         $query = QueryFilter::getInstance($argument);
 
+        if ($argName != $key) {
```

Comment:
How about we always do this and eliminate this check here and the check in QueryFilter.php?